File: lab2/Python (Zad 2.2-2.6, 2.8)/deserialize_json.py
# -*- coding: utf-8 -*-
"""
deserialize json
"""
import json

class DeserializeJson:
    # konstruktor
    def __init__(self, filename):
        print("let's deserialize something...")
        tempdata = open(filename, encoding="utf8")
        self.data = json.load(tempdata)

    # przykładowe statystyki
    # def somestats(self):
    #     example_stat = 0
    #     for dep in self.data:
    #         if dep['typ_JST'] == 'GM' and dep['Województwo'] == 'dolnośląskie': example_stat += 1
    #         print('liczba urzędów miejskich w województwie dolnośląskim: ' + ' ' + str(example_stat))

    # zadanie 2.3.3
    # modyfikacja funkcji somestats do wyświetlenia statystyki z każdego województwa
    def somestats(self):
        stats = {}
        for dep in self.data:
            jst_type = dep['typ_JST']
            wojewodztwo = dep['Województwo']
            if wojewodztwo not in stats:
                stats[wojewodztwo] = {}
            if jst_type not in stats[wojewodztwo]:
                stats[wojewodztwo][jst_type] = 0
            stats[wojewodztwo][jst_type] += 1

        for wojewodztwo, jst_types in stats.items():
            print('\nStatystyki dla województwa ' + wojewodztwo + ':')
            for jst_type, count in jst_types.items():
                print('\t- liczba urzędów typu ' + jst_type + ': ' + str(count))

File: lab2/C# (Zad 2.7)/IS_Lab2_JSON/DeserializeJson.cs
namespace IS_Lab2_JSON;

using System.Text.Json;

public class DeserializeJson
{
    public List<JST> data { get; set; }

    // konstructor
    public DeserializeJson(string filename)
    {
        string json = File.ReadAllText(filename);
        this.data = JsonSerializer.Deserialize<List<JST>>(json)!;
    }

    // przykładowe statystyki
    public void somestats()
    {
        Console.WriteLine("let's deserialize something...\n");

        Dictionary<string, Dictionary<string, int>> stats = new Dictionary<string, Dictionary<string, int>>();

        foreach (var dep in this.data)
        {
            if (!stats.ContainsKey(dep.Województwo))
                stats[dep.Województwo] = new Dictionary<string, int>();

            if (!stats[dep.Województwo].ContainsKey(dep.typ_JST))
                stats[dep.Województwo].Add(dep.typ_JST, 0);

            stats[dep.Województwo][dep.typ_JST] += 1;
        }

        foreach (var wojewodztwo in stats)
        {
            Console.WriteLine("Statystyki dla województwa " + wojewodztwo.Key + ":");

            foreach (var typJST in wojewodztwo.Value)
                Console.WriteLine("\t- liczba urzędów typu " + typJST.Key + ": " + typJST.Value);

            Console.WriteLine();
        }
        Console.WriteLine("all data has been deserialized!");
    }
}
